src/qt/sendcoinsdialog.cpp

```diff
@@ -791,8 +791,13 @@ void SendCoinsDialog::useAvailableBalance(SendCoinsEntry* entry)
     // Include watch-only for wallets without private key
     m_coin_control->fAllowWatchOnly = model->wallet().privateKeysDisabled() && !model->wallet().hasExternalSigner();
 
+    // Same behavior as send: if we have selected coins, only obtain their available balance.
+    // future, introduce a checkbox to customize this value.
```

Comment:
Basically to follow the same procedure as the send process (`SendCoinsDialog::PrepareSendText`), it's is at line 288-290.

Currently, the GUI disables the automatic selection of "other inputs" when the user manually selected coins (the process will only use those). But, in the future, that could also be customizable. The user might want to let the wallet add new inputs, even when the user had selected some, if there aren't enough funds to cover for the tx target amount.

Reply:
I see. I think it's better to drop that comment and instead open a GUI issue. The checkbox could say "Add more coins as needed" to match the send RPC behaviour. And it can point to the line `!coin_control.HasSelected();` that needs to be changed to implement this.

Reply:
np about creating a gui issue for this.

still, I'm not fan of the link to this particular line because this isn't the only place that has to be modified to introduce the feature. Plus, code changes over time, so the issue could easily get behind, which would mean more maintenance burden for us.

Reply:
pushed